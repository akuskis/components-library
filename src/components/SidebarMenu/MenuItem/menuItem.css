.menu-item {
  text-decoration: none;
  color: grey;
  margin: 10px 0 20px;
  text-align: left;
  padding-right: 10px;
  line-height: 0.1em;
}

.menu-item_active {
  color: black;
  border-bottom: 1px solid #000;
}

.menu-item_active span {
  padding-right: 10px;

  background: #fff;
}
