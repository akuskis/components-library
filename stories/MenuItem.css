.sign-in-story {
  width: 100%;

  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-between;
}

.sign-in-story__form {
  width: 400px;
  margin-top: 20px;
}
