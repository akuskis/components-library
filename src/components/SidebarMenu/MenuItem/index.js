// @flow

import React from 'react';
import { NavLink } from 'react-router-dom';

import './menuItem.css';

type Props = {
  title: string,
  routePath: string
};

const MenuItem = ({ title, routePath }: Props) => (
  <NavLink to={routePath} className="menu-item" activeClassName="menu-item_active">
    <div>{title}</div>
    <div className="menu-item__trailing-line" />
  </NavLink>
);

export default MenuItem;
