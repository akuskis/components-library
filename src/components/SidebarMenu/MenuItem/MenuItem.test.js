import React from 'react';
import { shallow } from 'enzyme';

import MenuItem from '.';

describe('<MenuItem />', () => {
  it('should render', () => {
    const component = shallow(<MenuItem title="Title" routePath="/route" />);

    expect(component).toMatchSnapshot();
  });
});
