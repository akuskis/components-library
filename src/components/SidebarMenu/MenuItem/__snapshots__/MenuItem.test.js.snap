// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> should render 1`] = `
<NavLink
  activeClassName="menu-item_active"
  aria-current="page"
  className="menu-item"
  to="/route"
>
  <span>
    Title
  </span>
</NavLink>
`;
